birdset/datamodule/embeddings_datamodule.py

```diff
@@ -0,0 +1,293 @@
+from typing import Union
+from birdset.modules.models.embedding_abstract import EmbeddingModel
+import torch
+import torchaudio
+import os
+
+
+log = pylogger.get_pylogger(__name__)
+
+@dataclass
+class EmbeddingModuleConfig(NetworkConfig):
+    """
+    A dataclass that makes sure the model inherits from EmbeddingClassifier.
+
+    """
+    model: Union[EmbeddingModel, torch.nn.Module] = None # Model for extracting the embeddings
+
+class EmbeddingDataModule(BaseDataModuleHF):
+    def __init__(
+            self,
+            dataset: DatasetConfig = DatasetConfig(),
+            loaders: LoadersConfig = LoadersConfig(),
+            transforms: BirdSetTransformsWrapper = BirdSetTransformsWrapper(),
+            mapper: None = None,
+            k_samples: int = 0,
+            val_batches: int = None, # Should val set be created
+            test_ratio: float = 0.5, # Ratio of test set if val set is also created
+            low_train: bool = False, # If low train set is used
+            embedding_model: EmbeddingModuleConfig = EmbeddingModuleConfig(),
+            average: bool = True,
+            gpu_to_use: int = 0
+    ):
+        super().__init__(
+            dataset=dataset,
+            loaders=loaders,
+            transforms=transforms
+        )
+        self.device = torch.device(f'cuda:{gpu_to_use}')
+        self.k_samples = k_samples
+        self.val_batches = val_batches
+        self.test_ratio = test_ratio
+        self.low_train = low_train
+        self.average = average
+        self.id_to_label = defaultdict(str)
+        self.embedding_model_name = embedding_model.model_name
+        self.embedding_model = embedding_model.model.to(self.device) # Move Model to GPU
+        self.embedding_model.eval()  # Set the model to evaluation mode
+        self.sampling_rate = embedding_model.sampling_rate
+        self.max_length = embedding_model.length
+        self.disk_save_path = os.path.join(
+            self.dataset_config.data_dir,
+            f"{self.dataset_config.dataset_name}_processed_{self.dataset_config.seed}_{self.embedding_model_name}_{self.k_samples}_{self.average}_{self.val_batches}_{self.low_train}_{self.sampling_rate}_{self.max_length}",
+        )
+        print(f"Using embedding model:{embedding_model.model_name} (Sampling Rate:{self.sampling_rate}, Window Size:{self.max_length})")
+
+    def _ksamples(self, dataset):
+        """
+        Use k_samples > 0 if you want control over amount of samples per class. The rest is used for validation and testing.
+        If test_ratio == 1 then no validation set even if k_samples == 0!
+        """
+        if self.k_samples > 0:
+            print(f">> Selecting {self.k_samples} Samples per Class this may take a bit...")
+            merged_data = concatenate_datasets([dataset['train'], dataset['valid'], dataset['test']])
+
+            # Shuffle the merged data
+            merged_data.shuffle() #? Check if this is affected by the public seed
```

Comment:
Prepend comment with `TODO`, as this is quite important, create an issue in our board for this!